Controllers/AddWeatherToIntervention.swift

```diff
@@ -0,0 +1,194 @@
+//
+//  AddWeatherToIntervention.swift
+//  Clic&Farm-iOS
+//
+//  Created by Jonathan DE HAAY on 07/09/2018.
+//  Copyright © 2018 Ekylibre. All rights reserved.
+//
+
+import UIKit
+import CoreData
+
+extension AddInterventionViewController {
+
+  // MARK: - Initialization
+
+  func initializeWeatherView() {
+    cloudyButton.layer.borderColor = UIColor.lightGray.cgColor
+    cloudyButton.layer.borderWidth = 2
+    cloudyButton.layer.cornerRadius = 5
+
+    cloudyPassageButton.layer.borderColor = UIColor.lightGray.cgColor
+    cloudyPassageButton.layer.borderWidth = 2
+    cloudyPassageButton.layer.cornerRadius = 5
+
+    fogyButton.layer.borderColor = UIColor.lightGray.cgColor
+    fogyButton.layer.borderWidth = 2
+    fogyButton.layer.cornerRadius = 5
+
+    rainButton.layer.borderColor = UIColor.lightGray.cgColor
+    rainButton.layer.borderWidth = 2
+    rainButton.layer.cornerRadius = 5
+
+    rainFallButton.layer.borderColor = UIColor.lightGray.cgColor
+    rainFallButton.layer.borderWidth = 2
+    rainFallButton.layer.cornerRadius = 5
+
+    snowButton.layer.borderColor = UIColor.lightGray.cgColor
+    snowButton.layer.borderWidth = 2
+    snowButton.layer.cornerRadius = 5
+
+    stormButton.layer.borderColor = UIColor.lightGray.cgColor
+    stormButton.layer.borderWidth = 2
+    stormButton.layer.cornerRadius = 5
+
+    sunnyButton.layer.borderColor = UIColor.lightGray.cgColor
+    sunnyButton.layer.borderWidth = 2
+    sunnyButton.layer.cornerRadius = 5
+  }
+
+  func defineWeathers() -> [UIButton] {
+    var weathers = [UIButton]()
+
+    weathers.append(cloudyButton)
+    weathers.append(cloudyPassageButton)
+    weathers.append(fogyButton)
+    weathers.append(rainButton)
+    weathers.append(rainFallButton)
+    weathers.append(snowButton)
+    weathers.append(stormButton)
+    weathers.append(sunnyButton)
+    return weathers
+  }
+
+  // MARK: - Actions
+
+  func hideWeatherItems(_ state: Bool) {
+    cloudyButton.isHidden = state
+    cloudyPassageButton.isHidden = state
+    fogyButton.isHidden = state
+    rainButton.isHidden = state
+    rainFallButton.isHidden = state
+    snowButton.isHidden = state
+    stormButton.isHidden = state
+    sunnyButton.isHidden = state
+    windSpeedTextField.isHidden = state
+    temperatureTextField.isHidden = state
+  }
+
+  @IBAction func collapseOrExpandWeatherView(_ sender: Any) {
```

Comment:
You can avoid using condition. See how I did it here:
https://github.com/ekylibre/zero-ios-v3/blob/master/Controllers/AddIrrigation.swift#L55

Reply:
And since we don't need animations for expandButtons, we no longer need "CGFloat.pi - 3.14159"